src/api/c/array.cpp

```diff
@@ -94,11 +94,11 @@ af_err af_create_handle(af_array *result, const unsigned ndims, const dim_t * co
                         const af_dtype type)
 {
     try {
-        af_array out;
+        af_array out = 0;
         AF_CHECK(af_init());
 
-        dim4 d((size_t)dims[0]);
```

Comment:
Add an assert before the for loop to check if `dims` is not `NULL` when `ndils > 0`